freqtrade/exchange/exchange.py

```diff
@@ -1270,40 +1270,64 @@ def extract_cost_curr_rate(self, order: Dict) -> Tuple[float, str, Optional[floa
 
     # Historic data
 
-    def get_historic_ohlcv(self, pair: str, timeframe: str,
-                           since_ms: int, is_new_pair: bool = False) -> List:
+    def get_historic_ohlcv(
+        self,
+        pair: str,
+        timeframe: str,
+        since_ms: int,
+        is_new_pair: bool = False,
+        price: Optional[str] = None
+    ) -> List:
         """
         Get candle history using asyncio and returns the list of candles.
         Handles all async work for this.
         Async over one pair, assuming we get `self.ohlcv_candle_limit()` candles per call.
         :param pair: Pair to download
         :param timeframe: Timeframe to get data for
         :param since_ms: Timestamp in milliseconds to get history from
+        :param price: "mark" if retrieving the mark price cnadles, "index" for index price candles
         :return: List with candle (OHLCV) data
         """
         return asyncio.get_event_loop().run_until_complete(
-            self._async_get_historic_ohlcv(pair=pair, timeframe=timeframe,
-                                           since_ms=since_ms, is_new_pair=is_new_pair))
+            self._async_get_historic_ohlcv(
+                pair=pair,
+                timeframe=timeframe,
+                since_ms=since_ms,
+                is_new_pair=is_new_pair,
+                price=price
```

Comment:
for the downloading itself, i think you'll have to go "wider" (changing how these are stored) - as currently, you'd be overwriting / combining these candles with already existing candles, which is for sure problematic / a deal breaker.

Reply:
> as currently, you'd be overwriting / combining these candles with already existing candles

Which part overwrites/combines them?

Reply:
well everything? at the moment - there's no code trying to write this dat to different files.

without that - it'll be combined - and as the format matches - there'll be no difference - so new candles for this will be merged with other candles - so you'll end up with data you can't use, as the data is 100% combined (your file `XRP_USDT_5m.json` will contain a mix of spot candles, and mark candles - assuming you run both one after the other with a few minutes gap between).

Reply:
Wouldn't I just need to find the part that declares the file name, and name it something like `mark_XRP_USDT_5m.json`

Reply:
you can find what you're looking for in freqtrade/data/history.
It's however not one file - but 3 (one base class, and the json and hdf5 implementations of the datahandlers).

Reply:
I think I might have fixed it, I haven't tested it out